contributors/devel/api_changes.md

```diff
@@ -538,30 +565,56 @@ of api objects - this is to improve the overall system performance.
 
 The auto-generated code resides with each versioned API:
 
-   - `pkg/api/<version>/types.generated.go`
-   - `pkg/apis/extensions/<version>/types.generated.go`
+   - `staging/src/k8s.io/api/<group>/<version>/types.generated.go`
 
 To regenerate them run:
 
 ```sh
 hack/update-codecgen.sh
 ```
 
-## Making a new API Group
+## Making a new API Version
 
 This section is under construction, as we make the tooling completely generic.
 
-At the moment, you'll have to make a new directory under `pkg/apis/`; copy the
-directory structure from `pkg/apis/authentication`. Add the new group/version to all
-of the `hack/{verify,update}-generated-{deep-copy,conversions,swagger}.sh` files
-in the appropriate places--it should just require adding your new group/version
-to a bash array.  See [docs on adding an API group](adding-an-APIGroup.md) for
-more.
-
-Adding API groups outside of the `pkg/apis/` directory is not currently
-supported, but is clearly desirable. The deep copy & conversion generators need
-to work by parsing go files instead of by reflection; then they will be easy to
-point at arbitrary directories: see issue [#13775](http://issue.k8s.io/13775).
+If you are adding a new API version to an existing group, you can copy the
+structure of the existing `pkg/apis/<group>/<existing-version>` and
+`staging/src/k8s.io/api/<group>/<existing-version>` folders.
+
+Due to the fast changing nature of the project, the following content is probably out-dated:
+* You can control if the version is enabled by default by update
+[pkg/master/master.go](https://github.com/kubernetes/kubernetes/blob/v1.8.0-alpha.2/pkg/master/master.go#L381).
+* You must add your new version to
+  [pkg/apis/<group>/install/install.go](https://github.com/kubernetes/kubernetes/blob/v1.8.0-alpha.2/pkg/apis/apps/install/install.go).
+* You must add the new version to
+  [hack/lib/init.sh#KUBE_AVAILABLE_GROUP_VERSIONS](https://github.com/kubernetes/kubernetes/blob/v1.8.0-alpha.2/hack/lib/init.sh#L53).
+* You must add the new version  to
+  [cmd/libs/go2idl/go-to-protobuf/protobuf/cmd.go](https://github.com/kubernetes/kubernetes/blob/v1.8.0-alpha.2/cmd/libs/go2idl/go-to-protobuf/protobuf/cmd.go#L64)
```

Comment:
this will soon be `staging/src/k8s.io/kube-gen/cmd/go-to-protobuf/protobuf/cmd.go`. Clearly, we don't want to keep the apigroup list in there medium term.